assets/js/plugin-check-admin.js

```diff
@@ -207,4 +211,89 @@
 
 		return data;
 	}
+
+	/**
+	 * Renders results for each check on the page.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} results The results object.
+	 */
+	function renderResults( results ) {
+		const { errors, warnings } = results;
+
+		// Render errors and warnings for files.
+		for ( const file in errors ) {
+			if ( warnings[ file ] ) {
+				renderFileResults( file, errors[ file ], warnings[ file ] );
+				delete warnings[ file ];
+			} else {
+				renderFileResults( file, errors[ file ], [] );
+			}
+		}
+
+		// Render remaining files with only warnings.
+		for ( const file in warnings ) {
+			renderFileResults( file, [], warnings[ file ] );
+		}
+	}
+
+	/**
+	 * Renders the file results table.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} file     The file name for the results.
+	 * @param {Object} errors   The file errors.
+	 * @param {Object} warnings The file warnings.
+	 */
+	function renderFileResults( file, errors, warnings ) {
+		const tableTemplate = wp.template( 'plugin-check-results-table' );
+		const rowTemplate = wp.template( 'plugin-check-results-row' );
+		const index = Date.now();
+
+		// Render the file table.
+		resultsContainer.innerHTML += tableTemplate( { file, index } );
+		const resultsTable = document.getElementById(
+			'plugin-check__results-body-' + index
+		);
+
+		// Loop over each result by the line, column and messages.
+		for ( const line in errors ) {
+			for ( const column in errors[ line ] ) {
+				for ( let i = 0; i < errors[ line ][ column ].length; i++ ) {
+					const message = errors[ line ][ column ][ i ].message;
+					const code = errors[ line ][ column ][ i ].code;
+
+					resultsTable.innerHTML += rowTemplate( {
+						line,
+						column,
+						file,
+						type: 'ERROR',
+						message,
+						code,
+					} );
+				}
+			}
+		}
```

Comment:
Repeated code again. There only two params that are different here are `errors` and the row type 'ERROR'. 

This should be very simple to change to a function.